### DISCUSSION TITLE
Storage Requirements

### DISCUSSION THREAD

**Discussion by happ104610** [htpcBeginner/docker-traefik#150] [Q&A] [answered]
I'd like to get this running in on of my Proxmox environments, how much hard drive storage would you allocate on a VM to get everything running? Thanks!

**Comment by SimpleHomelab** [comment, marked as answer]
Ok. I have this running on Proxmox as well.

I have 2 disks:

- 32 GB disk for the primary (root partition) - backup enabled. 18 GB of this is full at this point.
- 64 GB disk for things that I can afford to lose - backup disabled. Things that I can afford to lose include /var/lib/docker folder (which typically grows huge over time, Plex Metadata and cache, Radarr/Sonarr/Lidarr MediaCovers. All of these together right now make up about 40 GB of the 64 total. https://github.com/htpcBeginner/docker-traefik/discussions/147

The reason I have it this is ways is that I only backup the 32 GB VM disk. Backups are close 6 GB each and I have a full backup of the system without the things (in 64 GB disk) that can be recreated if lost.

**Reply by happ104610** [reply]
Thanks! I was thinking of a similar setup for the root partition.... I like to keep my storage for the local proxmox storage as low as possible and utilize my NAS(TrueNAS) storage for my media and cached files but I'm concerned about the performance of these cached files on a remote server. Is your 64GB disk (non-backup) also on your local Proxmox server or remote NAS storage? If remote storage, any performance issues you have run into? 

I've got a 2 Proxmox servers in a cluster and being able to migrate VMs between the 2 is incredibly convenient during server maintenance/testing.

**Reply by SimpleHomelab** [reply]
My 64 GB is a separate VM disk on the local proxmox server. I only use NAS for media storage (no performance issues). I also have media on Google Drive, which is rclone cached on the NAS and that works fine too. But for Plex cache, metadata, etc. I would try and use Proxmox local storage. In my case it is on the primary NVMe drive and it is blazing fast.

I am trying to get into clustering but ran into issues (I am new to it). Your 2 proxmox servers are in the same network? I have a proxmox server at home and one in a data center. Wondering if I can cluster them. I would like to do what you have going with Proxmox clustering.

**Reply by happ104610** [reply]
Both of my nodes are on the same network. I've got several NFS shares (for Backups & ISOs) that are defined at the Datacenter level in Proxmox and its my understanding that these defined shares must be available on each cluster node. In this setup, I would have to make the NAS available externally available to a cloud datacenter which I cannot imagine doing at this point. I do like the idea of backing up my cloud VM servers to my local storage though.  My Proxmox cluster is used mostly to migrate VMs to a working machine for server maintenance, testing, and proxmox upgrades. The machines don't need to be the identical but I do think local storage should be very similar between the 2 machines.
